File: management-domain/Sales-department/Backend/Java/notification-service/src/main/java/com/gogidix/sales/notification/shared/exception/NotFoundException.java
package com.gogidix.sales.notification.shared.exception;

import lombok.Getter;

/**
 * Not Found Exception
 * Thrown when a requested resource is not found
 */
@Getter
public class NotFoundException extends RuntimeException {

    private final String resource;
    private final String identifier;

    public NotFoundException(String resource, String identifier) {
        super(String.format("%s not found: %s", resource, identifier));
        this.resource = resource;
        this.identifier = identifier;
    }

    public NotFoundException(String resource) {
        super(String.format("%s not found", resource));
        this.resource = resource;
        this.identifier = null;
    }
}

File: management-domain/Finance-department/Backend/Nodes/finance-automation-service/invoice-processing-service/src/shared/exceptions/not-found.exception.ts
import { HttpStatus } from '@nestjs/common';
import { BaseException } from './base.exception';

/**
 * Exception thrown when a requested resource is not found
 */
export class NotFoundException extends BaseException {
  constructor(resource: string, identifier: string, context?: Record<string, unknown>) {
    super(
      `${resource} with identifier '${identifier}' not found`,
      HttpStatus.NOT_FOUND,
      'RESOURCE_NOT_FOUND',
      { resource, identifier, ...context },
    );
  }
}
